### ISSUE TITLE
Changing configuration via USB-Serial interface

### ISSUE THREAD

**Issue by FFF27** [astlouys/Pico-Green-Clock#10]
This is inspired by #9 because I don't want to enter my WiFi password over the 3 button interface.

Either a command line, or a simple menu where you can select from numbered options.

**Comment by astlouys** [comment]
Hello Frank,

I understand that entering letters / numbers / other symbols using the 3 clock buttons doesn't make great sense...

So, you're suggesting being able to enter them using "CDC" (serial protocol over USB line) and an external PC, or as an alternative, to have, say, 5 different combinations "user name / password" from which to select with the clock buttons. Did I get you idea correctly ?

As you mentioned, this is inspired by the idea of supporting NTP and could be done in relation with this other development.

**Comment by FFF27** [comment]
Yes, I did mean CDC serial stdin.

My general idea is as follows: 
* Somewhere in the main loop use `int getchar_timeout_us (uint32_t timeout_us )` to detect if a key was pressed
* If the key wasn't `c`, print on the console `Press c to configure`
* If the key was `c`, suspend normal operation and normal debug output
* Display a menu that's something like this:
```
     1) Time options
     2) WiFi options
     3) (etc...)

     0) Exit menu
```


Besides USB CDC or UART, the other ways to deploy a config seem to be: 
* set the password at compile time, every user will have to compile it themselves
* having the wifi start out in Access Point mode, the user can connect to the AP and access a web interface. Popular method with  commercial IoT devices.
* uploading a config file to littleFS. preferred method on ESP32, but seems to require Arduino IDE
* config script that runs on Windows/Linux, outputs a custom uf2 file that writes directly to flash.  The [uf2 format](https://github.com/microsoft/uf2) is really simple and can do partial overwrites anywhere in flash memory.  

Especially the custom uf2 seems to be more viable than I initially assumed.

**Comment by astlouys** [comment]
Hi Frank,

Your last method seems interesting and very simple to implement ("config script that runs on Windows / Linux").

I'm not sure how many users are comfortable to configure CDC serial over USB. In fact, the Pico Green Clock being... a clock (!), I would expect that some users are not tech savvy with good computer knowledge. And I didn't have enough feedback from users to make my mind on that matter...

Whatever the case, your last method is something that I've already done (in another environment) many years ago and it proved to be a KISS solution ("Keep It Short and Simple"). In any case, I think the first step is NTP implementation. When the main code is ready, it shouldn't be too difficult to add the login credentials.

Or course, you also mentioned other potential uses (your menu above) that are not covered with user name / password...

**Comment by astlouys** [comment]
Hi Frank,

I'm not planning to implement Green-Clock configuration through CDC USB. Since the configuration is now kept in flash, you will usually go through the clock configuration only once, so I don't think this option would be very useful.

However, if you want to give it a try, you may want to take a look at the Pico-Remote-Analyzer in my repositories. This Firmware uses CDC USB for both input and output and could be used as an example.
